### PR TITLE
Add sqrt builtin function

### PR DESCRIPTION
**Pull request by hugomg**
I'm going to be using it on some benchmarks.

The diff is easier to read if you only look at the first commit (7316848 ).

### CODE REVIEW

File: pallene/builtins.lua [hunk 1]
```diff
@@ -4,6 +4,7 @@ local builtins = {}
 
 for lua_name, typ in pairs({
     ["io.write"]     = types.T.Function({types.T.String()}, {}),
+    ["sqrt"]         = types.T.Function({types.T.Float()}, {types.T.Float()}),
```
Comment (gligneul) [inline pallene/builtins.lua]: Could we rename it to `math.sqrt`?

File: spec/coder_spec.lua [hunk 2]
```diff
@@ -1082,6 +1082,30 @@ describe("Pallene coder /", function()
...
+    describe("sqrt builtin", function()
+        setup(compile([[
+            function square_root(x: float): float
+                return sqrt(x)
+            end
+        ]]))
+
+        it("works on positive numbers", function()
+            run_test([[
+                assert(1.0 == test.square_root(1.0))
```
Comment (gligneul) [inline spec/coder_spec.lua]: It might be nice to test some corner cases such as nan, inf and negative numbers.

File: pallene/builtins.lua (hunk 1 above)
Comment (hugomg) [reply pallene/builtins.lua]: Good idea. Is more consistent with io.write.

File: spec/coder_spec.lua (hunk 2 above)
Comment (hugomg) [reply spec/coder_spec.lua]: Added NaN and inf. We already had tests for negative numbers.
